### PR TITLE
refactor: align get_exports of CommonJsExportRequireDependency

### PR DESCRIPTION
**Pull request by LingyuCoder**
## Summary

Align get_exports and get_referenced_exports of CommonJsExportRequireDependency to make tree shaking works

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by LingyuCoder** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/web-infra-dev/rspack/5588?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #5542** <a href="https://app.graphite.dev/github/pr/web-infra-dev/rspack/5542?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5543** <a href="https://app.graphite.dev/github/pr/web-infra-dev/rspack/5543?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5502** <a href="https://app.graphite.dev/github/pr/web-infra-dev/rspack/5502?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5588** <a href="https://app.graphite.dev/github/pr/web-infra-dev/rspack/5588?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/web-infra-dev/rspack/5588?utm_source=stack-comment).

**Comment by LingyuCoder** [comment]
!bench

**Comment by LingyuCoder** [comment]
!diff

File: crates/rspack_core/src/init_fragment.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@ pub enum InitFragmentKey {
   HarmonyImport(String),
```
Comment (LingyuCoder) [inline crates/rspack_core/src/init_fragment.rs:27]: !diff

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**